scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -1410,6 +1412,97 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
       context.parallelize[Unit](Seq(())).reifySideInputAsValues(view(coll)).values
     )
 
+  // =======================================================================
+  // Resource operations
+  // =======================================================================
+
+  def resourceCollectFn[R, U: Coder](resource: R, resourceType: ResourceType)(
+    pfn: PartialFunction[(R, T), U]
+  ): DoFn[T, U] =
+    new DoFnWithResource[T, U, R] {
```

Comment:
The error you are getting is because these can't be anonymous classes. We need to move this to a `CollectFnWithResource` class.